File: crates/dashboard-model/src/victron_controller/dashboard/v0_1_0/debug_full_charge.rs


#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, serde::Serialize, serde::Deserialize)]
pub enum DebugFullCharge {
    Forbid,
    Force,
    None_,
}

impl DebugFullCharge {
    pub fn parse(s: &str) -> Result<Self, String> {
        match s {
            "Forbid" => Ok(DebugFullCharge::Forbid),
            "Force" => Ok(DebugFullCharge::Force),
            "None_" => Ok(DebugFullCharge::None_),
            _ => Err(format!("Unknown variant: {}", s)),
        }
    }

    pub fn all() -> Vec<DebugFullCharge> {
        vec![
            DebugFullCharge::Forbid,
            DebugFullCharge::Force,
            DebugFullCharge::None_,
        ]
    }
}

impl std::fmt::Display for DebugFullCharge {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            DebugFullCharge::Forbid => write!(f, "Forbid"),
            DebugFullCharge::Force => write!(f, "Force"),
            DebugFullCharge::None_ => write!(f, "None_"),
        }
    }
}

impl std::str::FromStr for DebugFullCharge {
    type Err = String;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Self::parse(s)
    }
}

impl crate::baboon_runtime::BaboonBinCodecIndexed for DebugFullCharge {
    fn index_elements_count(_ctx: &crate::baboon_runtime::BaboonCodecContext) -> u16 {
        0
    }
}

impl crate::baboon_runtime::BaboonBinEncode for DebugFullCharge {
    fn encode_ueba(&self, _ctx: &crate::baboon_runtime::BaboonCodecContext, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        match self {
            DebugFullCharge::Forbid => crate::baboon_runtime::bin_tools::write_byte(writer, 0)?,
            DebugFullCharge::Force => crate::baboon_runtime::bin_tools::write_byte(writer, 1)?,
            DebugFullCharge::None_ => crate::baboon_runtime::bin_tools::write_byte(writer, 2)?,
        }
        Ok(())
    }
}

impl crate::baboon_runtime::BaboonBinDecode for DebugFullCharge {
    fn decode_ueba(_ctx: &crate::baboon_runtime::BaboonCodecContext, reader: &mut dyn std::io::Read) -> Result<Self, Box<dyn std::error::Error>> {
        let tag = crate::baboon_runtime::bin_tools::read_byte(reader)?;
        match tag {
            0 => Ok(DebugFullCharge::Forbid),
            1 => Ok(DebugFullCharge::Force),
            2 => Ok(DebugFullCharge::None_),
            _ => Err(format!("Unknown enum variant tag: {}", tag).into()),
        }
    }
}

File: web/src/model/victron_controller/dashboard/DebugFullCharge.ts
// @ts-nocheck
import {BaboonCodecContext, BaboonBinWriter, BinTools, BaboonBinReader, Lazy} from '../../BaboonSharedRuntime'

export enum DebugFullCharge {
    Forbid = "Forbid",
    Force = "Force",
    Auto = "Auto"
}

export const DebugFullCharge_values: ReadonlyArray<DebugFullCharge> = [
    DebugFullCharge.Forbid,
    DebugFullCharge.Force,
    DebugFullCharge.Auto
] as const;

export function DebugFullCharge_parse(s: string): DebugFullCharge {
    const found = DebugFullCharge_values.find(v => v === s);
    if (found === undefined) {
        throw new Error("Unknown DebugFullCharge variant: " + s);
    }
    return found;
}

export class DebugFullCharge_UEBACodec {
    public encode(ctx: BaboonCodecContext, value: DebugFullCharge, writer: BaboonBinWriter): unknown {
        if (this !== DebugFullCharge_UEBACodec.lazyInstance.value) {
          return DebugFullCharge_UEBACodec.lazyInstance.value.encode(ctx, value, writer)
        }
    
        switch (value) {
            case "Forbid": BinTools.writeByte(writer, 0); break;
                case "Force": BinTools.writeByte(writer, 1); break;
                case "Auto": BinTools.writeByte(writer, 2); break;
            default: throw new Error("Unknown enum variant: " + value);
        }
    }
    
    public decode(ctx: BaboonCodecContext, reader: BaboonBinReader): DebugFullCharge {
        if (this !== DebugFullCharge_UEBACodec .lazyInstance.value) {
            return DebugFullCharge_UEBACodec.lazyInstance.value.decode(ctx, reader)
        }
    
        const tag = BinTools.readByte(reader);
        switch (tag) {
            case 0: return "Forbid" as DebugFullCharge;
                case 1: return "Force" as DebugFullCharge;
                case 2: return "Auto" as DebugFullCharge;
            default: throw new Error("Unknown enum variant tag: " + tag);
        }
    }

    public static readonly BaboonDomainVersion = '0.3.0'
    public baboonDomainVersion() {
        return DebugFullCharge_UEBACodec.BaboonDomainVersion
    }
    public static readonly BaboonDomainIdentifier = 'victron_controller.dashboard'
    public baboonDomainIdentifier() {
        return DebugFullCharge_UEBACodec.BaboonDomainIdentifier
    }
    public static readonly BaboonTypeIdentifier = 'victron_controller.dashboard/:#DebugFullCharge'
    public baboonTypeIdentifier() {
        return DebugFullCharge_UEBACodec.BaboonTypeIdentifier
    }

    protected static lazyInstance = new Lazy(() => new DebugFullCharge_UEBACodec())
    public static get instance(): DebugFullCharge_UEBACodec {
        return DebugFullCharge_UEBACodec.lazyInstance.value
    }
}